### PR TITLE
EOS-13637 Fix s3recovery tool to seamlessly handle unexpected failure…

### PR DESCRIPTION
**Pull request by shanbhagsv1**
…s during tool run

### CODE REVIEW

File: s3recovery/s3recovery/s3recoverybase.py [hunk 1]
```diff
@@ -297,10 +317,6 @@ def dry_run(self, index_name, index_id, index_id_replica, union_result, recover_
 
             self.perform_validation(key, metadata_value, replica_value, union_result)
 
-            # Perform cleanup for existing indices during recovery
-            if (recover_flag):
```
Comment (nileshgovande) [inline s3recovery/s3recovery/s3recoverybase.py:301]: recover_flag is redundant - let us remove

File: s3recovery/s3recovery/s3recovercorruption.py [hunk 2]
```diff
@@ -31,6 +31,32 @@ def __init__(self):
         super(S3RecoverCorruption, self).create_logger_directory()
         super(S3RecoverCorruption, self).create_logger("S3RecoverCorruption")
 
+    def cleanup_entries(self, index_id):
```
[the remark this answers is not in the document]
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recovercorruption.py]: Done

File: s3recovery/s3recovery/s3recovercorruption.py [hunk 3]
```diff
@@ -31,6 +31,32 @@ def __init__(self):
...
+
+        """
+        list_index_response = self.list_index(index_id)
+        data_as_dict = self.parse_index_list_response(list_index_response)
+        key_list = list(data_as_dict.keys())
+        for key in key_list:
+            if key not in self.common_keys:
+                self.kv_api.delete(index_id,key)
+
+    def cleanup_metadata_entries(self, index_id):
```
[the remark this answers is not in the document]
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recovercorruption.py]: Done

File: s3recovery/s3recovery/s3recoverybase.py [hunk 4]
```diff
@@ -100,7 +117,8 @@ def put_kv(self, index_id, key, value):
         :value: Value to be inserted
 
         """
-        self.kv_api.put(index_id, key, value)
+        status, response = self.kv_api.put(index_id, key, value)
+        self.check_response(status, "put", response, index_id, key)
 
     def perform_cleanup(self, key, index_id, index_id_replica):
```
[the remark this answers is not in the document]
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recoverybase.py]: Done

File: s3recovery/s3recovery/s3recoverybase.py (hunk 1 above)
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recoverybase.py:301]: Done

File: s3recovery/s3recovery/s3recovercorruption.py [hunk 5]
```diff
@@ -31,6 +31,33 @@ def __init__(self):
...
+        Performs cleanup of stale data entries
+
+        """
+        list_index_response = self.list_index(index_id)
+        data_as_dict = self.parse_index_list_response(list_index_response)
+        key_list = list(data_as_dict.keys())
+        for key in key_list:
+            if key not in self.common_keys:
+                status, response = self.kv_api.delete(index_id, key)
+                super(S3RecoverCorruption, self).check_response(status, "delete", response, index_id, key)
```
Comment (amitwac2608) [inline s3recovery/s3recovery/s3recovercorruption.py:45]: we can return the status and response from this function, to help caller to check the status, if needed ?

File: s3recovery/s3recovery/s3recoverybase.py [hunk 6]
```diff
@@ -91,29 +91,34 @@ def s3recovery_log(self, log_level, msg):
             elif log_level == "critical":
                 self.logger.critical(msg)
 
-    def put_kv(self, index_id, key, value):
+    def check_response(self, status, request, response, index_id, key):
         """
-        Puts the given key-value to corresponding index-id
+        Validate the response received from server
```
Comment (amitwac2608) [inline s3recovery/s3recovery/s3recoverybase.py:96]: s3server

**Review by amitwac2608 — APPROVED** [review]
Looks good, gave couple of comments, see if they make sense.

File: s3recovery/s3recovery/s3recovercorruption.py (hunk 5 above)
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recovercorruption.py:45]: Return is not required as we check status and response as a part of the check_response call.

File: s3recovery/s3recovery/s3recoverybase.py (hunk 6 above)
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recoverybase.py]: Done

**Comment by shanbhagsv1** [comment]
Build Job http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/11/ Success

**Review by nileshgovande — APPROVED** [review]

**Pull request merged**